cmd/setup-etcd-environment/run.go

```diff
@@ -86,9 +105,54 @@ func runRunCmd(cmd *cobra.Command, args []string) error {
 		out = f
 	}
 
-	if err := writeEnvironmentFile(map[string]string{
-		"DISCOVERY_SRV": runOpts.discoverySRV,
-	}, out, true); err != nil {
+	export := make(map[string]string)
+	etcdName := os.Getenv("ETCD_NAME")
+	if etcdName == "" {
+		return fmt.Errorf("environment variable ETCD_NAME has no value")
+	}
+	var e EtcdScaling
+	if runOpts.pivot {
+		clientConfig, err := rest.InClusterConfig()
+		if err != nil {
+			panic(err.Error())
+		}
+		client, err := kubernetes.NewForConfig(clientConfig)
+		if err != nil {
+			return fmt.Errorf("error creating client: %v", err)
+		}
+		duration := 10 * time.Second
+		// wait forever for success and retry every duration interval
+		wait.PollInfinite(duration, func() (bool, error) {
+			result, err := client.CoreV1().ConfigMaps("openshift-etcd").Get("scaling-lock", metav1.GetOptions{})
```

Comment:
At least in this usage, "scaling-lock" seems misnamed.
Perhaps "known-members"?

Reply:
yeah, this configmap originally was a distributed lock similar to the leader election process. But I did not find that to be necessary will consider renaming.